controllers/hcloudmachine_controller.go

```diff
@@ -83,6 +84,18 @@ func (r *HCloudMachineReconciler) Reconcile(ctx context.Context, req reconcile.R
 
 	log = log.WithValues("HCloudMachine", klog.KObj(hcloudMachine))
 
+	if utils.UpdateFinalizer(hcloudMachine, infrav1.DeprecatedHCloudMachineFinalizer, infrav1.HCloudMachineFinalizer) {
+		// Finalizers got updated. Write new object to api-server and reconcile again
+		err = r.Update(ctx, hcloudMachine)
```

Comment:
we patch the machine

Reply:
I updated the code so that the util method is not needed.